### ISSUE TITLE
MPV will not open on a Raspberry Pi 4

### ISSUE THREAD

**Issue by ithiliat** [d-rez/vitadock-cec#1]
Hi,
I went through all the steps and the installation went ok, but it doesn't seem to be working for me.
The screen on my Vita just blinks, and nothing is happening on the screen.

I'm using a Pi4b if that might have anything to do with it? I was using the VLC image before, but as you stated that it would not work I downloaded Crash's Pi4 image  (which had both MPV and VLC configured). I tested my Vita prior to running the commands, and everything was working fine (I made sure it had MPV working).

I checked the logs as I was connecting/disconnecting my Vita. It looks fine to me. It does not seem to register when i switch inputs on my TV though?

INFO:root:Init: CEC library
INFO:root:Vita connected
INFO:root:Starting stream
INFO:root:Vita connected
INFO:root:Vita disconnected
INFO:root:Terminating stream
INFO:root:Vita disconnected
INFO:root:Vita connected
INFO:root:Starting stream
INFO:root:Vita connected

One other thing. After I ran the last command, this came up:

File "/home/pi/vitadock-cec/vitadock-cec.py", line 125, in <module>
    logging.info("Current TV state: " + "on" if tv.is_on() else "off")
OSError: Power status not found

Quick update:
I just wanted to let you know that I initially was trying to use this image:
https://www.reddit.com/r/VitaDock/comments/faddi3/vitadock_mpv_console_beta_release/

But my Pi would not boot from this.

Thanks

**Comment by d-rez** [comment]
Thanks for testing and the feedback. 

Unfortunately I don't have a Pi4 lying around to wipe and check the Pi4 MPV image. There might be some more changes in it.

Since Pi4 has 2x HDMI the CEC commands might be failing, but I have no way to check that right now. Could you try the other display output maybe? Are you sure your TV supports HDMI-CEC and that the feature is enabled?

Regarding MPV only can you try the script at 'testing/test-mpv.py'? If it does nothing (no logo, no stream) after a few seconds, then maybe another build of MPV needs to be used

Again, thanks for taking your time, afraid Pi4 support is not something I'll be focusing on right now, I'll keep this issue open. 

It would probably be more effective to add VLC support instead and figure out what's causing CEC to not function on Pi4

**Comment by ithiliat** [comment]
Only one of the HDMI ports seem to be working. I don't know if Crash has made any adjustments on this image, beacause I seem to recall both working when using the VLC image before. But im not 100% sure.

My TV has CEC enabled and was working before when I was using the VLC image.

I tried again today while tailing the log and nothing happens when I connect my Vita anymore. The only two lines in the log are:

INFO:root:Init: Ading CEC callbacks
INFO:root:Init: CEC library

I ran 'testing/test-mpv.py', but that doesn't seem to do anything. I tried it multiple times and in various different ways (both with my Vita connected and connecting it after executing the script).

I was thinking that maybe I could download the raspbian image and try to install it manually instead of using the prebuilt image, since your MPV test script didn't work.

Thanks

**Comment by ithiliat** [comment]
Ok, so I have made some progress.
I re-applied the MPV image and ran all the commands once again. When I tried different HDMI ports on my Pi, I noticed that I needed to restart it in order for it to display on my TV. So when I tried a different HDMI port the logs were looking a lot better! It was logging whenever I connected or disconnected my Vita. However, MPV still did not work.
I did some research and found out that you could force HDMI out on the Pi (in case it doesn't recognize that a display is connected). So I added **hdmi_force_hotplug=1** to my /boot/config.txt. Now I can see the following happening in my logs when I turn on or off my TV (nothing is happening if I switch between different sources and if I connect my Vita while on a different source):

INFO:root:Starting stream
INFO:root:Vita connected
INFO:root:TV off
INFO:root:Terminating stream
INFO:root:TV off
INFO:root:Vita disconnected
INFO:root:Vita disconnected
INFO:root:TV on
INFO:root:Resuming previous stream
INFO:root:Starting stream
INFO:root:Vita connected
INFO:root:Vita connected
INFO:root:Vita disconnected
INFO:root:Terminating stream
INFO:root:Vita disconnected
INFO:root:TV on
INFO:root:Resuming previous stream
INFO:root:Starting stream
INFO:root:TV off
INFO:root:Terminating stream

MPV still doesn't start though. When I reinstalled everything I noticed this during the installation:

dpkg: warning: downgrading mpv from 0.32.0-1 to 0.29.0-1
(Reading database ... 180094 files and directories currently installed.)
Preparing to unpack .../mpv_0.29.0-1_armhf.deb ...
Unpacking mpv (0.29.0-1) over (0.32.0-1) ...
dpkg: warning: unable to delete old directory '/usr/local/share/applications': Directory not empty
dpkg: warning: unable to delete old directory '/usr/local/share': Directory not empty
Setting up mpv (0.29.0-1) ...
Processing triggers for man-db (2.8.5-2) ...

Also, I ran the test-mpv.py again after this, still nothing.

Thanks in advance!

**Comment by d-rez** [comment]
> I noticed that I needed to restart it in order for it to display on my TV

Yep that's how Pi4 works with HDMI output, if you boot it without one / with specific one, you have to reboot for any changes

I can mark two things thanks to your effort:

1) I need to check my script on a newer VD build. Clearly MPV is being downgraded and that's not good. More recompiling for me... yay? I'll take a look. Can't promise anything for Pi4 hybrid image though as I don't have a spare Pi4 lying around (all my Pi4's are in use atm). But that's alright since the current build isn't too reliable anyway

2. Your TV's CEC handling is different to mine lol. It works (good!) but not fully (only power events). If you want, can you switch to VD and run `testing/test-cec.py` ? Press ENTER a few times (for visual separation, switch to another source, wait for messages to stop. Press ENTER a few times again and switch back to the dock source, then attach/paste, I'll take a look how to improve source change logic. Hopefully something will work for all TVs

**Comment by ithiliat** [comment]
Yeah, sorry. My bad..


I ran `testing / test-cec.py`. I hope it helps you in some way.

**VD source:**

4 ({'initiator': 0, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 135, 'parameters': b'\x00\x90>', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 0, 'destination': 1, 'ack': False, 'eom': False, 'opcode': 144, 'parameters': b'\x00', 'opcode_set': True, 'transmit_timeout': 1000},)


**Different source:**

4 ({'initiator': 0, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 134, 'parameters': b'@\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 8, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 130, 'parameters': b'@\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 0, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 134, 'parameters': b'@\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 8, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 130, 'parameters': b'@\x00', 'opcode_set': True, 'transmit_timeout': 1000},)


**VD source:**

4 ({'initiator': 0, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 134, 'parameters': b'0\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
64 (True, 1)
4 ({'initiator': 0, 'destination': 1, 'ack': False, 'eom': False, 'opcode': 0, 'parameters': b'\x8e\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 0, 'destination': 1, 'ack': False, 'eom': False, 'opcode': 143, 'parameters': b'', 'opcode_set': True, 'transmit_timeout': 1000},)
4 ({'initiator': 0, 'destination': 15, 'ack': False, 'eom': False, 'opcode': 134, 'parameters': b'0\x00', 'opcode_set': True, 'transmit_timeout': 1000},)
64 (True, 1)

Thanks for all the work you put into this, I really appreciate it!

**Comment by d-rez** [comment]
Gonna be useful, thanks. I'll take a look sometime this week
